src/common/snippets/src/pass/fuse_transpose_brgemm.cpp

```diff
@@ -17,73 +17,88 @@
 namespace ngraph {
 namespace snippets {
 namespace pass {
+
 const std::set<std::vector<int>> FuseTransposeBrgemm::supported_cases = {{0, 2, 1, 3}};
+
+bool FuseTransposeBrgemm::is_supported_transpose(const Output<Node>& transpose_port) {
+    const auto transpose_node = transpose_port.get_node_shared_ptr();
+    // it's safe to do so because of the patterns we used. alternatively we can do it through pattern_values_map
+    const auto& constant = as_type_ptr<ngraph::opset1::Constant>(transpose_node->get_input_node_shared_ptr(1));
+    // if Transpose in and out layout is not empty => something was already fused on this port
+    if (!utils::get_node_output_layout(transpose_node).empty() ||
+        !utils::get_node_output_layout(transpose_node->get_input_node_shared_ptr(0)).empty())
+        return false;
+    const auto& transpose_order = constant->cast_vector<int>();
+    // todo: this limitation is due to the fact that offsets are calculated in Kernel, and the only way
```

Comment:
This todo seems obsolete. Remove?

Reply:
I think we cannot remove it since We support fused `Transpose` only after `Parameter`. This todo is about general cases when `Transpose` can be before `MatMul` (in which he will be fused) and after several nodes